### PR TITLE
feat: Add subcommands to server

### PR DESCRIPTION
**Pull request by mkmik**
Part of #917; the goal there is to add a bunch of commands like:

```console
$ influxdb_iox server remote set 23 http://10.1.1.23:8080
```

To that end, I turned the `server` into a "subcommand group" layer, and added a `run` command that actually runs the server:

```console
$ influxdb_iox server run
```

The rest of the PR contains some simplifications and cleanups.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -170,33 +168,28 @@ fn main() -> Result<(), std::io::Error> {
...
+            Command::Server(config) => {
                 // Note don't set up basic logging here, different logging rules apply in server
                 // mode
-                if let Err(e) = commands::server::command(logging_level, config).await {
+                if let Err(e) = commands::server::command(logging_level, *config).await {
                     eprintln!("Server command failed: {}", e);
                     std::process::exit(ReturnCode::Failure as _)
                 }
             }
-            None => {
```
Comment (mkmik) [inline src/main.rs:195]: In #940 I found a convoluted way to require at least one command (involving `SubcommandRequiredElseHelp`), but I didn't notice that all I had to do was to remove the Option at line 86:

```
command: Option<Command>,
```

**Review by alamb — COMMENTED** [review]
I tried this out locally and I was having some problems -- namely `server run` doesn't actually run 🤔 


```
alamb@ip-10-0-0-124 influxdb_iox % ./target/debug/influxdb_iox server run
./target/debug/influxdb_iox server run
Mar 11 10:07:52.669  WARN influxdb_iox::influxdb_ioxd: NO PERSISTENCE: using Memory for object storage
Mar 11 10:07:52.670  WARN influxdb_iox::influxdb_ioxd: server ID not set. ID must be set via the INFLUXDB_IOX_ID config or API before writing or querying data.
Server command failed: Server error
```

When I gave it all the info to prevent warnings it still doesn't seem to run
```
alamb@ip-10-0-0-124 influxdb_iox % ./target/debug/influxdb_iox server run --object-store=file --data-dir=$HOME/.influxdb_iox  --writer-id=42
<nfluxdb_iox server run --object-store=file --data-dir=$HOME/.influxdb_iox  --writer-id=42
Mar 11 10:14:32.614  INFO influxdb_iox::influxdb_ioxd: Using File for object storage
Server command failed: Server error
```

**Comment by mkmik** [comment]
works for me;

it seems that the problem is in the error message propagation, because `Server command failed: Server error` is not telling us what's wrong on your side.

**Comment by mkmik** [comment]
fwiw the error message propagation issue is a regression in main:

```console
$ git checkout main
$ cargo run -- server --object-store=file --data-dir=/bad/tmp/abcd
Server command failed: Server error
```

**Comment by mkmik** [comment]
@alamb can you please try again, I fixed the `thiserror` error message percolation, so perhaps we can see what's wrong

**Review by alamb — APPROVED** [review]
@mkmik  --- I tried again at:


```
    run     Runs in server mode
alamb@ip-10-0-0-124 influxdb_iox % cargo run -- server run
cargo run -- server run
    Finished dev [unoptimized + debuginfo] target(s) in 0.38s
     Running `target/debug/influxdb_iox server run`
Mar 11 11:02:08.692  WARN influxdb_iox::influxdb_ioxd: NO PERSISTENCE: using Memory for object storage
Mar 11 11:02:08.693  WARN influxdb_iox::influxdb_ioxd: server ID not set. ID must be set via the INFLUXDB_IOX_ID config or API before writing or querying data.
Server command failed: Server error: Unable to bind to listen for gRPC requests on 127.0.0.1:8082: Address already in use (os error 48)

``` 
🤦 

... and When I stopped my other server it worked great!


Thanks for fixing the error message

**Comment by mkmik** [comment]
> Thanks for fixing the error message

@alamb thanks for hitting the regression! :-)

**Pull request merged**